js/controller.js

```diff
@@ -830,12 +830,16 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
 
     onVcPlay: function(event, source) {
       this.state.currentVideoId = source;
-      if (this.state.adWasPaused && this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai) {
-        this.state.adPauseDuration = Date.now() - this.state.adPausedTime;
-        //we calculate new ad end time, based on the time that the ad was paused.
-        this.state.adEndTime = this.state.adEndTime + this.state.adPauseDuration; //milliseconds
-        this.state.adWasPaused = false;
-        this.state.adPauseDuration = 0;
+      if (this.state.currentAdsInfo.currentAdItem.ssai) {
```

Comment:
@ipadilla4 I think you need the null checks that are in the nested if statement here. This check will cause an exception if `this.state.currentAdsInfo.currentAdItem` or `this.state.currentAdsInfo` is null. Is the solution to just set `this.state.currentVideoId` to `OO.VIDEO.MAIN`? If so, then I think we could remove this outer if and move `this.state.currentVideoId = OO.VIDEO.MAIN` inside the previous `if (this.state.adWasPaused && this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai)` statement

Reply:
Yes this is needed only for ssai ads. I forgot to add the null checks to the outer if. We do need two ifs because `onVcPlay` method is called twice, one for `"main"` source and one for `"ads"` . When parameter `source` is `"ads"`, `this.state.adWasPaused` will be false.